src/contrib/cluster/Akka.Cluster.Tools.Tests/PublishSubscribe/DistributedPubSubMediatorSpec.cs

```diff
@@ -58,47 +59,27 @@ public void DistributedPubSubMediator_can_be_activated_in_child_actor()
         }
 
         [Fact]
-        public void DistributedPubSubMediator_should_get_dead_letter()
+        public async Task DistributedPubSubMediator_should_send_messages_to_dead_letter()
         {
-            EventFilter.Exception<NullReferenceException>().Expect(0, () =>
+            await EventFilter.DeadLetter<object>().ExpectAsync(10, TimeSpan.FromMinutes(3), () =>
             {
+                var mediator = DistributedPubSub.Get(Sys).Mediator;
                 var actor = Sys.ActorOf((dsl, context) =>
                 {
-                    IActorRef mediator = null;
-                    ICancelable cancelable = null;
-                    dsl.OnPreStart = actorContext =>
-                    {
-                        Sys.EventStream.Subscribe(context.Self, typeof(DeadLetter));
-                        mediator = DistributedPubSub.Get(actorContext.System).Mediator;
-                        mediator.Tell(new Subscribe("pub-sub", context.Self));
-                    };
-
-                    dsl.Receive<string>(s => s.Equals("UnSub"), (s, actorContext) =>
-                    {
-                        mediator.Tell(new Unsubscribe("pub-sub", context.Self));
-                    }); 
-                    dsl.Receive<SubscribeAck>((s, actorContext) =>
-                    {
-                        actorContext.System.Log.Info($"{s.Subscribe.Topic}");
-                        cancelable = context.System.Scheduler.ScheduleTellOnceCancelable(TimeSpan.FromSeconds(5), context.Self, "UnSub", context.Self);
-                    }); 
-                    dsl.Receive<PublishTopic>((s, actorContext) =>
-                    {
-                        mediator.Tell(new Publish("pub-sub", "Good"));
-                    }); 
-                    dsl.Receive<DeadLetter>((s, actorContext) =>
-                    {
-                        actorContext.System.Log.Info($"Received deadletter {s.Message}");
-                    }); 
-                    dsl.Receive<UnsubscribeAck>((s, actorContext) =>
+                }, "childActor");
+                mediator.Tell(new Subscribe("pub-sub", actor));
+                _ = ExpectMsg<SubscribeAck>();
+                mediator.Tell(new Unsubscribe("pub-sub", actor));
+                _ = ExpectMsg<UnsubscribeAck>();
+                Sys.Scheduler.Advanced.ScheduleOnce(TimeSpan.FromMinutes(2.50), () => 
```

Comment:
The default `removed-time-to-live` is 2 minutes - that is when pruning takes effect if there are no subscribers to a Topic. Hence, the reason for the `TimeSpan.FromMinutes(2,50)`